asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc

```diff
@@ -1,8 +1,240 @@
 [id="changes-in-vertx-jdbc-client_{context}"]
 = Changes in {VertX} JDBC client
 
+From {VertX} {v4}, the JDBC client supports SQL client. The SQL common module has also been merged in JDBC client, that is, `io.vertx:vertx-sql-common` merged in `io.vertx:vertx-jdbc-client` module. You will have to remove the `io.vertx:vertx-sql-common` dependency file because `io.vertx:vertx-jdbc-client` will include it. With the merging of SQL common client, all the database APIs have been consolidated into the JDBC client.
+
+In {VertX} {v4}, the SQL client has been updated to include the following clients:XXX
+
+* Reactive PostgreSQL client. In earlier releases, it included a reactive PostgreSQL client.
+
+* Reactive MySQL client
+* Reactive DB2 client
+* Existing JDBC client that includes the JDBC client API and the SQL client API.
+
+The reactive implementations use the database network protocols. This makes them resource-efficient.
+
+JDBC calls to database are blocking calls. The JDBC client uses worker threads to make these calls non-blocking. XXX
+
 The following section describes the changes in {VertX} JDBC client.
 
-== Merged Vert.x SQL common module in JDBC client module
+== Creating a pool
+
+In {VertX} {v4}, you can create a pool using the JDBC client APIs. In earlier releases, you could create only clients. You could not create pools.
+
+The following example shows how to create a client in {VertX} {v3x}.
+
+[source,java]
+----
+// 3.x
+SQLClient client = JDBCClient.create(vertx, jsonConfig);
+----
+
+The following example shows how to create a pool in {VertX} {v4}.
+
+[source,java]
+----
+// 4.x
+JDBCPool pool = JDBCPool.pool(vertx, jsonConfig);
+----
+
+NOTE: Though the {VertX} {v3x} APIs are supported in {VertX} {v4}, it is recommended that you use the new JDC client APIs in your applications.
+
+A pool enables you to perform simple queries. You do not need to manage connections for simple queries. However, for complex queries or multiple queries, you must manage your connections.
+
+The following example shows how to manage connections for queries in {VertX} {v3x}.
+
+[source,java]
+----
+// 3.x
+client.getConnection(res -> {
+  if (res.succeeded()) {
+    SQLConnection connection = res.result();
+    // very important! do not forget to return the connection
+    connection.close();
+  } else {
+    // Failed to get connection - deal with it
+  }
+});
+----
+
+The following example shows how to manage connections for queries in {VertX} {v4}.
+
+[source,java]
+----
+// 4.x
+pool
+  .getConnection()
+  .onFailure(e -> {
+    // failed to get a connection
+  })
+  .onSuccess(conn -> {
+    // very important! don't forget to return the connection
+    conn.close();
+  });
+----
+
+== Support for Typsesafe Config XXX
```

Comment:
@pmlopes Does this section read correct?

Reply:
:+1: